app/src/main/java/org/mozilla/rocket/urlinput/SearchPortalViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package org.mozilla.rocket.urlinput
+
+import android.arch.lifecycle.MediatorLiveData
+import android.arch.lifecycle.ViewModel
+
+class SearchPortalViewModel(repository: SearchPortalRepository) : ViewModel() {
+
+    val searchPortalObservable = MediatorLiveData<ArrayList<SearchPortal>>()
+    private val liveGlobal = repository.fetchGlobal()
+    private val liveLocale = repository.fetchLocale()
+
+    init {
+        searchPortalObservable.addSource(liveGlobal) {
+            mergePortals()
+        }
+        searchPortalObservable.addSource(liveLocale) {
+            mergePortals()
+        }
+    }
+
+    private fun mergePortals() {
+        val result = ArrayList<SearchPortal>()
+        if (liveGlobal.value != null) {
+            result.addAll(liveGlobal.value!!)
```

Comment:
I think we should start to avoid!! in production code (though we have some already)